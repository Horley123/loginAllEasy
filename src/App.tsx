import React from 'react';
import {StatusBar} from 'react-native';
import {SafeAreaProvider, SafeAreaView} from 'react-native-safe-area-context';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import AppProvider from './hooks';
import Routes from './routes';
import {Backdrop} from './components/Backdrop';
import TooltipeError from './components/TooltipeError';
import {GluestackUIProvider} from '@gluestack-ui/themed';
import {config} from '@gluestack-ui/config';
import useThemeStore from './store/useThemeStore';

function App(): React.JSX.Element {
  const {theme} = useThemeStore();
  return (
    <SafeAreaProvider style={{justifyContent: 'center'}}>
      <SafeAreaView
        style={{
          flex: 1,
        }}>
        <GestureHandlerRootView style={{flex: 1}}>
          <GluestackUIProvider config={theme}>
            <AppProvider>
              <StatusBar
                translucent
                backgroundColor="transparent"
                barStyle="dark-content"
              />
              <Routes />
              <Backdrop />
              <TooltipeError />
            </AppProvider>
          </GluestackUIProvider>
        </GestureHandlerRootView>
      </SafeAreaView>
    </SafeAreaProvider>
  );
}

export default App;
