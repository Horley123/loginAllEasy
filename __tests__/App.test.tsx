// src/App.test.tsx
import React from 'react';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';
import App from './../src/App';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({navigate: jest.fn()}),
}));

jest.mock('react-native-gesture-handler', () => {
  const React = require('react');
  const {View} = require('react-native');
  return {
    GestureHandlerRootView: ({style, children}: any) =>
      React.createElement(View, {style}, children),
  };
});

jest.mock('react-native-safe-area-context', () => {
  const React = require('react');
  const {View} = require('react-native');
  return {
    SafeAreaProvider: ({children}: any) =>
      React.createElement(View, null, children),
    SafeAreaView: ({style, children}: any) =>
      React.createElement(View, {style}, children),
  };
});

jest.mock('@gluestack-ui/themed', () => ({
  GluestackUIProvider: ({children}: any) => children,
}));
jest.mock('@gluestack-ui/config', () => ({config: {}}));

jest.mock('./../src/hooks', () => {
  const React = require('react');
  return () =>
    ({children}: {children: React.ReactNode}) =>
      children;
});
jest.mock('./../src/routes', () => {
  const React = require('react');
  return () => React.createElement(React.Fragment, null);
});
jest.mock('./../src/components/Backdrop', () => {
  const React = require('react');
  return {Backdrop: () => React.createElement(React.Fragment, null)};
});
jest.mock('./../src/components/TooltipeError', () => {
  const React = require('react');
  return () => React.createElement(React.Fragment, null);
});

jest.mock('./../src/store/useThemeStore', () => () => ({theme: {}}));

describe('App', () => {
  it('deve renderizar sem crashar e bater snapshot', () => {
    const {toJSON} = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('App', () => {
  it('renderiza o texto corretamente', () => {
    const {getByText} = render(<Text>Bem-vindo</Text>);
    expect(getByText('Bem-vindo')).toBeTruthy();
  });
});
