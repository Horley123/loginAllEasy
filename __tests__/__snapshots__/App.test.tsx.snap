// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App deve renderizar sem crashar e bater snapshot 1`] = `
<View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
